spyder/widgets/projects/tests/test_projectdialog.py

```diff
@@ -10,6 +10,7 @@
 
 # Test library imports
 import pytest
+from unittest.mock import patch
```

Comment:
We use for the mock tests the monkeypatch fixture of pytest and the Mock class. So the import here should be something like [this](https://github.com/spyder-ide/spyder/blob/5abd2a27e1c30b9ab4656e5aa290e99ac15b5ccb/spyder/widgets/variableexplorer/tests/test_dataframeeditor.py#L19) that you can see in `..\spyder\spyder\widgets\variableexplorer\tests\test_dataframeeditor.py`

Reply:
First, not sure the exact reasons but our codebase uses ``monkeypatch`` rather than ``mock.patch``. Second, you need to actually import mock to use it. Third, in order to be compatible with Python 2 (your tests are failing there) you need:

```python
try:
    from unittest.mock import MagicMock
except ImportError:
    from mock import MagicMock  # Python 2
```